### ISSUE TITLE
Undefined offset when the data is over the page 

### ISSUE THREAD

**Issue by Frasaccordi** [dompdf/dompdf#1314]
i have loop for function for show data table for pdf stream. but there is always eror when the queries have a lot of data and should have second page. when the queries only have a little data an only need 1 page to sstream it, it works perfectly. only when the data is need second page i think. i dont know what exactly happen. is it second page process or i doing something wrong. but i try debug  update_row_group function with dd() in cellmap.php, when the data is enough with 1 page, this function is not called. please help i really stuck here and sorry for my bad english. hope this not make you confuse

This my Print Blade

````

<!DOCTYPE html>
<html>
<head>
 
<style>


table {

    width: 100%;
}

th, td {
    text-align: left;
    padding: 8px;
}

tr:nth-child(even){background-color: #f2f2f2}

th {
    background-color: #7D26CD;
    color: white;
}
.hidden {
    display: none;
}
.bg-green{
  background-color:#009900;
}
.title{
    text-align:center;
}

.titleHeader{
    width:20%;
}
.page { width: 100%; height: 100%; }

.breakNow { page-break-after:always; }

.page-break {
    page-break-after: always;
}

</style>


</head>
<body style="font-family:Helvetica;">

<h2 class="title">Pemantauan Kegiatan</h2>
<table>
  <tr>
    <th class="titleHeader">Kode dan Nama Program</th>
    <td>{{'['.$programnamenew[0]->pr_code.']'.$programnamenew[0]->pr_name}}</td>  
  </tr>
</table>

<table>
  <thead class="bg-light-blue cf" style="font-family:Helvetica;">
      <tr>
          <th class="title" style="width:5%;" rowspan="2">No</th>
          <th class="title" rowspan="2">Kode</th>
          <th class="title" style="width:40%;" rowspan="2">Nama Program</th>
          <th class="title bg-green" colspan="2"><span><i class="fa fa-bank">&nbsp;</i>Anggaran</span></th>
          <th class="title" colspan="2"><span><i class="fa fa-line-chart">&nbsp;</i>Kinerja</span></th>
      </tr>
      <tr>
          <th class="title">Pagu (Rp.)</th>
          <th class="title">Realisasi (Rp.)</th>
          <th class="title">%</th>
          <th class="title">Capaian</th>
      </tr>
  </thead>
  <tbody style="font-family:Helvetica; page-break-before: always;">
      <small class="hidden">0x3F{!!$listNumber = 1 !!}</small> 
     
      @for ($i = 0; $i < count($datas); $i++)
 
            @if($i==4)
            <div class="page-break"></div>
            @endif

     <tr data-toggle="collapse" data-target="#allkegiatandata" class="accordion-toggle">
            <td data-title="No" class="text-center">
                <h5>{{ $listNumber }}</h5></td>
            <td data-title="Kode" class="text-center">
                <h5>{{ $datas[$i]->ac_code }}</h5></td>
            <td data-title="Nama Kegiatan">
                <h5>{{ $datas[$i]->ac_name }}</h5></td>
                <td data-title="Anggaran Pagu" class="text-right small">
                        {{ number_format((float) $activity_budget[$listNumber - 1], 0, ',', '.') }}
                </td>
                <td data-title="Anggaran Realisasi" class="text-right small">
                    {{ number_format((float) $activity_real[$listNumber - 1], 0, ',', '.') }}
                </td>
                <td data-title="Kinerja Realisasi" class="text-right small">
                    <h5><span class="label bg-purple"><strong>
                    {{ number_format((float) $activity_real[$listNumber - 1]/(float) $activity_budget[$listNumber - 1]*100, 2, '.', ',') }} %
                    </strong></span></h5>
                </td>
                <td data-title="Kinerja - Capaian" class="text-center" style="font-family:DejaVu Sans; text-align:center; color:green;"><i>&#9899;</i></td>
            <!-- ANGGARAN -->
            <!-- pagu -->
            
           
        </tr>
        
      <small class="hidden">0x4F{!! $listNumber++ !!}</small> @endfor
     
      
      
  </tbody>
  
</table>
 

</body>
</html>


````

**Comment by bsweeney** [comment]
Do you have some actual output from the blade? Easier to debug if we can see what you're trying to render.

**Comment by bsweeney** [comment]
Your HTML structure is a bit off. You have some non-table element as children of the tbody element which is not valid per the HTML spec. This could be throwing off the rendering, though I believe those elements might be moved outside the table in later releases of dompdf.

What is the full text of the error?

**Comment by Frasaccordi** [comment]
Whoops, looks like something went wrong.
````
1/1
ErrorException in Cellmap.php line 800:
Undefined offset: 240
````

````
in Cellmap.php line 800
at HandleExceptions->handleError('8', 'Undefined offset: 240', 'D:\Mission\Projek\Holomoc\Playtext\Lemhannas\LEMHANNAS_MONEV\vendor\dompdf\dompdf\src\Cellmap.php', '800', array('group' => object(TableRowGroup), 'last_row' => object(Text), 'g_key' => '52', 'r_key' => '240')) in Cellmap.php line 800
at Cellmap->update_row_group(object(TableRowGroup), object(Text)) in TableRowGroup.php line 67
at TableRowGroup->split(object(TableRow), true) in AbstractFrameDecorator.php line 637
at AbstractFrameDecorator->split(null, true) in Page.php line 537
at Page->check_page_break(object(Text)) in Block.php line 741
at Block->reflow(object(TableCell)) in AbstractFrameDecorator.php line 801
at AbstractFrameDecorator->reflow(object(TableCell)) in TableCell.php line 91
at TableCell->reflow(null) in AbstractFrameDecorator.php line 801
at AbstractFrameDecorator->reflow() in TableRow.php line 46
at TableRow->reflow(null) in AbstractFrameDecorator.php line 801
at AbstractFrameDecorator->reflow() in TableRowGroup.php line 43
at TableRowGroup->reflow(null) in AbstractFrameDecorator.php line 801
at AbstractFrameDecorator->reflow() in Table.php line 488
at Table->reflow(object(Block)) in AbstractFrameDecorator.php line 801
at AbstractFrameDecorator->reflow(object(Block)) in Block.php line 738
at Block->reflow(null) in AbstractFrameDecorator.php line 801
at AbstractFrameDecorator->reflow() in Page.php line 133
at Page->reflow(null) in AbstractFrameDecorator.php line 801
at AbstractFrameDecorator->reflow() in Dompdf.php line 834
at Dompdf->render() in PDF.php line 187
at PDF->render() in PDF.php line 133
at PDF->output() in PDF.php line 170
at PDF->stream() in ActivityController.php line 155
at ActivityController->printActivity('3')
at call_user_func_array(array(object(ActivityController), 'printActivity'), array('id' => '3')) in Controller.php line 80
at Controller->callAction('printActivity', array('id' => '3')) in ControllerDispatcher.php line 146
at ControllerDispatcher->call(object(ActivityController), object(Route), 'printActivity') in ControllerDispatcher.php line 94
at ControllerDispatcher->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 52
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in Authenticate.php line 28
at Authenticate->handle(object(Request), object(Closure))
at call_user_func_array(array(object(Authenticate), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 136
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 103
at Pipeline->then(object(Closure)) in ControllerDispatcher.php line 96
at ControllerDispatcher->callWithinStack(object(ActivityController), object(Route), object(Request), 'printActivity') in ControllerDispatcher.php line 54
at ControllerDispatcher->dispatch(object(Route), object(Request), 'App\Http\Controllers\ActivityController', 'printActivity') in Route.php line 174
at Route->runController(object(Request)) in Route.php line 140
at Route->run(object(Request)) in Router.php line 724
at Router->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 52
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in VerifyCsrfToken.php line 64
at VerifyCsrfToken->handle(object(Request), object(Closure))
at call_user_func_array(array(object(VerifyCsrfToken), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 136
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in ShareErrorsFromSession.php line 49
at ShareErrorsFromSession->handle(object(Request), object(Closure))
at call_user_func_array(array(object(ShareErrorsFromSession), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 136
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in StartSession.php line 64
at StartSession->handle(object(Request), object(Closure))
at call_user_func_array(array(object(StartSession), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 136
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in AddQueuedCookiesToResponse.php line 37
at AddQueuedCookiesToResponse->handle(object(Request), object(Closure))
at call_user_func_array(array(object(AddQueuedCookiesToResponse), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 136
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in EncryptCookies.php line 59
at EncryptCookies->handle(object(Request), object(Closure))
at call_user_func_array(array(object(EncryptCookies), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 136
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 103
at Pipeline->then(object(Closure)) in Router.php line 726
at Router->runRouteWithinStack(object(Route), object(Request)) in Router.php line 699
at Router->dispatchToRoute(object(Request)) in Router.php line 675
at Router->dispatch(object(Request)) in Kernel.php line 246
at Kernel->Illuminate\Foundation\Http\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 52
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in CheckForMaintenanceMode.php line 44
at CheckForMaintenanceMode->handle(object(Request), object(Closure))
at call_user_func_array(array(object(CheckForMaintenanceMode), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 136
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 103
at Pipeline->then(object(Closure)) in Kernel.php line 132
at Kernel->sendRequestThroughRouter(object(Request)) in Kernel.php line 99
at Kernel->handle(object(Request)) in index.php line 54
at require_once('D:\Mission\Projek\Holomoc\Playtext\Lemhannas\LEMHANNAS_MONEV\public\index.php') in server.php line 21

`````

**Comment by Frasaccordi** [comment]
which part that not valid per the HTML spec? would you mark or recomment, please? thankyou

**Comment by bsweeney** [comment]
You have a `small` element at the start and end of your `tbody` element. You also have a page break `div` at the top. Neither of these elements should not be directly inside the `tbody` element. The `small` elements are hidden, so you can probably just take those out. Not sure what you're trying to achieve with the page-break `div` but you'll probably need to find another way to do it.

If I remove those three elements or enable the HTML5 parser (which fixes the HTML structure) the document renders without errors.

**Comment by Frasaccordi** [comment]
Works Perfectly! thankyou Buddy!

i use small hidden is for have a variable that i use for numbering and show some of array. so i try use $i which is increment from for looping to do that.

so DompPdf cannot cannot have Hidden element?

**Comment by bsweeney** [comment]
I can have hidden elements, and those are ignored during rendering. But the fact that they're hidden is not taken into account until after the document structure is parsed, which is why they could cause problems.

**Comment by bsweeney** [comment]
There are quite a few instances of this popping up (e.g. #1195, #1295, #1300). I don't know if we'll include a fix to this particular issue for 0.7.1, but likely 0.7.2 since it appears to be a common.
